File: Python/Move Zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        zero_index = 0
        for i in xrange(len(nums)):
            if nums[i] != 0:
                nums[i], nums[zero_index] = nums[zero_index], nums[i]
                zero_index += 1
        

File: cpp/Move Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int zero_index = 0;
        for (int i = 0; i < nums.size(); ++i) {
            if (nums[i]) {
                swap(nums[zero_index], nums[i]);
                zero_index++;
            }
        }
    }
};
